ui/src/main/java/edu/wpi/grip/ui/util/NodeControllerObservableListMap.java

```diff
@@ -0,0 +1,196 @@
+package edu.wpi.grip.ui.util;
+
+
+import com.google.common.collect.BiMap;
+import com.google.common.collect.HashBiMap;
+import edu.wpi.grip.ui.Controller;
+import javafx.application.Platform;
+import javafx.collections.FXCollections;
+import javafx.collections.ListChangeListener;
+import javafx.collections.ObservableList;
+import javafx.scene.Node;
+
+import java.util.*;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+public final class NodeControllerObservableListMap<C extends Controller, N extends Node> implements Map<C, N> {
+
+    private final ObservableList<N> nodesList;
+    private final BiMap<C, N> controllerNodeMap;
+
+    private final ListChangeListener listChangeListener = (ListChangeListener<N>) c -> {
+        while (c.next()) {
+            if (c.wasAdded() || c.wasRemoved()) {
+                throw new IllegalArgumentException("This list is now managed and should not be added or removed from");
```

Comment:
Document that this doesn't actually cause the add or remove operation to fail. It just **behind the scenes** calls the thread uncaught exception handler.